ipfsApi/exceptions.py

```diff
@@ -1,32 +1,126 @@
-"""Defines the skeleton for exceptions.
 """
+The class hierachy for exceptions is::
 
+    Error
+     +-- CommandError
+     |    +-- InvalidArguments
+     +-- EncoderError
+     |    +-- EncoderMissingError
+     |    +-- EncodingError
+     |    +-- DecodingError
+     +-- CommunicationError
+          +-- ProtocolError
+          +-- StatusError
+          +-- ErrorResponse
+          +-- ConnectionError
+          +-- TimeoutError
 
-class ipfsApiError(Exception):
-    """Base class for exceptions in this module."""
-    pass
+"""
 
 
-class InvalidCommand(ipfsApiError):
-    """Exception raised for an invalid command."""
+class Error(Exception):
+    """Base class for all exceptions in this module."""
     pass
 
 
-class InvalidArguments(ipfsApiError):
-    """Exception raised for invalid arguments."""
-    pass
+###############
+# commands.py #
```

Comment:
My thinking was to give a potential reader a general idea of "where is this used?", so that it's easier to figure out where this exception will probably originate from.